torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2286,7 +2309,8 @@ def _sharded_post_state_dict_hook(
                     num_devices_per_node=torch.cuda.device_count(),
                     pg=self.process_group
                 )  # type: ignore[assignment]
-        state_dict.pop(f"{prefix}{FLAT_PARAM}")
+        if not self._use_orig_params:
```

Comment:
nit: add a comment indicating that when use_orig_params, the flat param is not registered as an nn.Module parameter so it is not expected to be included in the state_dict.